src/query/service/src/pipelines/processors/transforms/transform_udf_script.rs

```diff
@@ -212,79 +254,39 @@ impl Transform for TransformUdfScript {
-
-            if let Some(runtime) = self.script_runtimes.get(&runtime_key) {
-                let result_batch = runtime.handle_execution(func, &input_batch, index)?;
-                self.update_datablock(func, result_batch, &mut data_block)?;
-            } else {
-                return Err(ErrorCode::UDFDataError(format!(
-                    "Failed to find runtime for function {:?} with key: {:?}",
-                    func.name, runtime_key
-                )));
-            }
+            let runtime = self.script_runtimes.get(&func.name).unwrap();
+            let result_batch = runtime.handle_execution(func, &input_batch)?;
+            self.update_datablock(func, result_batch, &mut data_block)?;
         }
         Ok(data_block)
     }
 }
 
 impl TransformUdfScript {
-    fn get_runtime_key(func: &UdfFunctionDesc) -> Result<String> {
-        let (lang, func_name) = match &func.udf_type {
-            UDFType::Script(UDFScriptCode { language: lang, .. }) => (lang, &func.func_name),
-            _ => {
-                return Err(ErrorCode::UDFDataError(format!(
-                    "Unsupported UDFType variant for function {:?}",
-                    func.name
-                )));
-            }
-        };
-
-        let runtime_key = format!("{}-{}", lang, func_name.trim());
-        Ok(runtime_key)
-    }
-
-    pub fn init_runtime(
-        funcs: &[UdfFunctionDesc],
-        runtime_num: usize,
-    ) -> Result<BTreeMap<String, Arc<ScriptRuntime>>> {
+    pub fn init_runtime(funcs: &[UdfFunctionDesc]) -> Result<BTreeMap<String, Arc<ScriptRuntime>>> {
         let mut script_runtimes: BTreeMap<String, Arc<ScriptRuntime>> = BTreeMap::new();
 
-        let start = std::time::Instant::now();
         for func in funcs {
-            let (&lang, code_opt) = match &func.udf_type {
-                UDFType::Script(UDFScriptCode { language, code, .. }) => {
-                    (language, Some(code.as_ref().as_ref()))
-                }
+            let code = match &func.udf_type {
+                UDFType::Script(code) => code,
                 _ => continue,
             };
 
-            let runtime_key = Self::get_runtime_key(func)?;
-            let runtime = match script_runtimes.entry(runtime_key.clone()) {
-                Entry::Occupied(entry) => entry.into_mut().clone(),
-                Entry::Vacant(entry) => {
-                    let new_runtime = ScriptRuntime::try_create(lang, code_opt, runtime_num)
-                        .map(Arc::new)
-                        .map_err(|err| {
-                            ErrorCode::UDFDataError(format!(
-                                "Failed to create UDF runtime for language {lang:?} with error: {err}",
-                            ))
-                        })?;
-                    entry.insert(new_runtime).clone()
-                }
+            if let Entry::Vacant(entry) = script_runtimes.entry(func.name.clone()) {
```

Comment:
key should include function language and function name, different language may have same function name

Reply:
```
create or replace FUNCTION echo1 () RETURNS INT64 LANGUAGE javascript HANDLER = 'echo' AS $$
export function echo() {
    return 1;
}
$$;

create or replace FUNCTION echo2 () RETURNS INT64 LANGUAGE javascript HANDLER = 'echo' AS $$
export function echo() {
    return 2;
}
$$;

select echo1(),echo2();
-- 1,1
```
the handler is not  unique.

Reply:
Actually, the name here is the handler name and not the function name, which is confusing. It's better to add some comments or change the field name to `handler_name`.